tox.ini

```diff
@@ -12,8 +13,23 @@ setenv =
     PYTHONDONTWRITEBYTECODE=1
     PYTHONWARNINGS=once
 deps =
-    django22: Django>=2.2,<3.0
     django30: Django>=3.0,<3.1
     django31: Django>=3.1,<3.2
     django32: Django>=3.2,<4.0
+    django40: Django>=4.0,<4.1
+    django41: Django>=4.1,<4.2
+    django42: Django>=4.2b1,<5.0
+    djangomain: https://github.com/django/django/archive/main.tar.gz
     -rrequirements-test.txt
+
+[testenv:py38-djangomain]
+ignore_outcome = true
+
+[testenv:py39-djangomain]
+ignore_outcome = true
+
+[testenv:py310-djangomain]
+ignore_outcome = true
+
+[testenv:py311-djangomain]
+ignore_outcome = true
```

Comment:
Not sure if testing against Django's development version is needed :thinking: What's the benefit, if any errors are discarded in any case?

Reply:
Yeah, it's definitely a bit extra. It can give you a sneak peek on how the library works with 5.0 or whatever is the current development build. There's also an important distinction here; the results are *ignored*, not *discarded*; the `djangomain` tests won't result in a failure if the tests won't pass, but you can still look at the results. [For example, here](https://github.com/City-of-Helsinki/django-helusers/actions/runs/4424441777/jobs/7758274159#step:6:231).

It's not needed, but it's mostly an extra step that can be useful. Of course, it doesn't seem like this particular library is in very active development, so the benefits might not really be there. Don't really have any objections for removing this.

Reply:
Perhaps there is a chance that those could be beneficial at some situation. Let's keep them. It appeared though that Django's development version requires at least Python 3.10, so is it OK to remove the `main` configuration from older Python versions?

Reply:
Yup, that's, ok. I'll do that 💪